worker/provisioner/container_initialisation_test.go

```diff
@@ -319,30 +347,49 @@ func (s *ContainerSetupSuite) assertContainerInitialised(c *gc.C, ctype instance
 		}
 	}
 
-	for _, pack := range packages {
+	for _, pack := range cont.packages {
 		cmd := <-s.aptCmdChan
-		expected := []string{
-			"apt-get", "--option=Dpkg::Options::=--force-confold",
-			"--option=Dpkg::options::=--force-unsafe-io", "--assume-yes", "--quiet",
-			"install"}
-		expected = append(expected, pack...)
+
+		expected := append(expected_initial, pack...)
 		c.Assert(cmd.Args, gc.DeepEquals, expected)
 	}
 }
 
 func (s *ContainerSetupSuite) TestContainerInitialised(c *gc.C) {
-	for _, test := range []struct {
-		ctype    instance.ContainerType
-		packages [][]string
-	}{
-		{instance.LXC, [][]string{
-			[]string{"--target-release", "precise-updates/cloud-tools", "lxc"},
-			[]string{"--target-release", "precise-updates/cloud-tools", "cloud-image-utils"}}},
-		{instance.KVM, [][]string{
-			[]string{"uvtool-libvirt"},
-			[]string{"uvtool"}}},
-	} {
-		s.assertContainerInitialised(c, test.ctype, test.packages, false)
+	var cont []ContainerInstance
+
+	current_os, _ := version.GetOSFromSeries(version.Current.Series)
+
+	switch current_os {
+	case version.CentOS:
+		cont = []ContainerInstance{
+			{instance.LXC, [][]string{
+				[]string{"lxc"},
+				[]string{"cloud-image-utils"}}},
+			{instance.KVM, [][]string{[]string{"uvtool-libvirt"}, []string{"uvtool"}}},
+		}
+	case version.Ubuntu:
+		cont = []ContainerInstance{
+			{instance.LXC, [][]string{
+				[]string{"--target-release", "precise-updates/cloud-tools", "lxc"},
+				[]string{"--target-release", "precise-updates/cloud-tools", "cloud-image-utils"}}},
+			{instance.KVM, [][]string{
+				[]string{"uvtool-libvirt"},
+				[]string{"uvtool"}}},
+		}
+	default:
+		cont = []ContainerInstance{
+			{instance.LXC, [][]string{
+				[]string{"--target-release", "precise-updates/cloud-tools", "lxc"},
+				[]string{"--target-release", "precise-updates/cloud-tools", "cloud-image-utils"}}},
+			{instance.KVM, [][]string{
+				[]string{"uvtool-libvirt"},
+				[]string{"uvtool"}}},
```

Comment:
I think default here and somewhere else should somehow signal that we're not supposed to run these tests on those machines. AFAIK these were skipped on windows. Also could you refactor these so you don't switch in both places?(you could pass cont as an argument for example).

You could also take a look at the juju/utils/packaging/commander package I think and you could try to extract the proper command from that package based on the series, without relying on strings.